modules/nlu/src/backend/stan/client.ts

```diff
@@ -1,134 +1,111 @@
-    const endpoint = `train/${stringId}/cancel`
-    return this._post(endpoint, { password })
+  public async listModels(credentials: Credentials): Promise<ListModelsResponseBody | ErrorResponse> {
+    return this._wrapWithTryCatch(async () => {
+      const endpoint = 'models/'
+      const { data } = await this._client.get(endpoint, { params: credentials })
+      return data
+    })
   }
 
-  public async hasModel(modelId: NLUEngine.ModelId, password: string): Promise<boolean> {
-    const stringId = NLUEngine.modelIdService.toString(modelId)
-    const endpoint = `exists/${stringId}`
-    const { exists } = await this._get(endpoint, { password })
-    return exists
+  public async pruneModels(credentials: Credentials): Promise<PruneModelsResponseBody | ErrorResponse> {
+    return this._wrapWithTryCatch(async () => {
+      const endpoint = 'models/prune'
+      const { data } = await this._client.get(endpoint, { params: credentials })
+      return data
+    })
   }
 
   public async detectLanguage(
-    utterance: string,
-    models: { modelId: NLUEngine.ModelId; password: string }[]
-  ): Promise<string> {
-    const { modelId, password } = models[0]
-    const stringId = NLUEngine.modelIdService.toString(modelId)
-    const endpoint = `detect-lang/${stringId}`
-    const { languages } = await this._post(endpoint, { utterances: [utterance], password })
-    return languages[0]
-  }
-
-  public async predict(utterance: string, modelId: NLUEngine.ModelId, password: string): Promise<PredictOutput> {
-    const stringId = NLUEngine.modelIdService.toString(modelId)
-    const endpoint = `predict/${stringId}`
-    const { predictions } = await this._post(endpoint, { utterances: [utterance], password })
-    return predictions[0]
+    detectLangRequestBody: DetectLangRequestBody
+  ): Promise<DetectLangResponseBody | ErrorResponse> {
+    return this._wrapWithTryCatch(async () => {
+      const endpoint = 'detect-lang'
+      const { data } = await this._client.post(endpoint, detectLangRequestBody)
+      return data
+    })
   }
 
-  private async _get(endpoint: string, queryParams: any) {
-    try {
-      const { data } = await this._client.get(endpoint, { params: queryParams })
-      const { success } = data
-      if (!success) {
-        this._throwError(data)
-      }
+  public async predict(
+    modelId: string,
+    predictRequestBody: PredictRequestBody
+  ): Promise<PredictResponseBody | ErrorResponse> {
+    return this._wrapWithTryCatch(async () => {
+      const endpoint = `predict/${modelId}`
+      const { data } = await this._client.post(endpoint, predictRequestBody)
       return data
-    } catch (err) {
-      const errMsg = `The following errored occured when calling standalone NLU: [${err}]`
-      this._logger.error(errMsg)
-    }
+    })
   }
 
-  private async _post(endpoint: string, body: any) {
+  private async _wrapWithTryCatch<T>(fn: () => Promise<T>) {
```

Comment:
Wouldn't a decorator be cleaner for this kind of stuff?

Reply:
good point, but I could not make it build, I kept having this error:
`[module-builder] Error transpiling file "src/backend/stan/client.ts"`

Reply:
You should use `"experimentalDecorators": true` in the `tsconfig.json`

Reply:
well, I'd rather not play with the module-builder tbh